ArduPlane/navigation.cpp

```diff
@@ -339,9 +339,13 @@ void Plane::update_fbwb_speed_height(void)
         }
 
 #if SOARING_ENABLED == ENABLED
-        if (g2.soaring_controller.is_active() && g2.soaring_controller.get_throttle_suppressed()) {
-            // we're in soaring mode with throttle suppressed
-            set_target_altitude_current();;
+        if (g2.soaring_controller.is_active()) {
+            if (g2.soaring_controller.get_throttle_suppressed()) {
+                // we're in soaring mode with throttle suppressed
+                set_target_altitude_current();
+            } else {
+                target_altitude.amsl_cm = 100*plane.g2.soaring_controller.get_alt_cutoff() + 1000 + AP::ahrs().get_home().alt;
```

Comment:
Uh - `+1000` ?

Even if there's a good reason - AFAICS that reason should be in a comment...

Reply:
Aka... we're allergic to magic numbers

Reply:
Added a comment. Let me know if there is a better way of handling this situation, I don't like magic numbers either

Reply:
Well, this is your library, so it's definitely not a blocker  - and the comment makes it crystal clear what's going on.

But yeah, magic numbers are usually bad :-)